app/src/main/java/net/wetfish/wetfish/ui/behaviors/FAMShrinkBehavior.java

```diff
@@ -0,0 +1,75 @@
+package net.wetfish.wetfish.ui.behaviors;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.View;
+import android.view.animation.Animation;
+import android.view.animation.AnimationUtils;
+import android.widget.RelativeLayout;
+
+import com.google.android.material.snackbar.Snackbar;
+
+import net.wetfish.wetfish.R;
+
+import androidx.coordinatorlayout.widget.CoordinatorLayout;
+import androidx.core.view.ViewCompat;
+
+/**
+ * Created by ${Michael} on 1/18/2019.
+ */
+
+public class FAMShrinkBehavior extends CoordinatorLayout.Behavior<RelativeLayout> {
+
+    private Context mContext;
+    private Animation mScaleDown;
+    private Animation mScaleUp;
+
+    public FAMShrinkBehavior(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        mContext = context;
+
+        mScaleDown = AnimationUtils.loadAnimation(mContext, R.anim.fam_scale_down);
+        mScaleUp = AnimationUtils.loadAnimation(mContext, R.anim.fam_scale_up);
+    }
+
+    @Override
+    public boolean layoutDependsOn(CoordinatorLayout parent, RelativeLayout child, View dependency) {
+        return dependency instanceof Snackbar.SnackbarLayout;
+    }
+
+    @Override
+    public boolean onDependentViewChanged(CoordinatorLayout parent, RelativeLayout child, View dependency) {
+        float translationY = Math.min(0, dependency.getTranslationY() - dependency.getHeight());
+        // Note that the RelativeLayout gets translated.
```

Comment:
Might want more comments on this new code, but it's good you do have some.

Reply:
I'll look it over and see if I can put some m ore comments on an upcoming commit. Thanks!